packages/types/src/package.ts

```diff
@@ -22,6 +22,103 @@ export interface Package {
    * The state exposed by this package.
    */
   state?: {
+    /**
+     * The configuration objects for all the fills exposed by this package.
+     */
+    fills?: {
+      [namespace: string]: {
+        [name: string]: Fill;
+      };
+    };
+
+    /**
+     * The rest of the state exposed by this package.
+     *
+     * TODO: Replace `any` with serializable objects and derived state.
```

Comment:
Good catch! I didn't add the TODO and I think I overlooked it. I'll fix that code.

Reply:
by the way, I think the "serializable objects" is referring to: https://github.com/frontity/frontity/blob/dev/packages/types/src/utils.ts#L76-L79 (I think) in case this was not clear.

Reply:
I'm afraid I wasn't able to define a `Packate["state"]` type using `Serializable` and `Derived`. :disappointed: 

I tried something similar to this (and other variations) without success:
```ts
type StateProps = {
  [prop: string]:
    | Serializable
    | Derived<Package, any>
    | Derived<Package, any, any>
    | StateProps
};
```

Other types start to complain, like `ResolveState` and `ResolveActions`, and some package definitions as well. I already spent too much time on this and I've started to think that it's not worth the time.

@michalczaplinski, maybe you want to take a look at this, just don't spend much time.

Reply:
I've had a quick look but was not able to get it to work so let's leave it as is :( 

Only FYI, I've noticed that the `Serializable` can maybe be simplified to just: 

```
export type Serializable =
  | Primitive
  | Serializable[]
  | { [key: string]: Serializable };
```

in the future.

Reply:
I guess we can try to simplify both `ResolveState` and `ResolveActions` in the future with some of the techniques explained by Titian in the video that Michal shared the other day: https://www.youtube.com/watch?v=wNsKJMSqtAk.

Reply:
Do we have a feature discussion or a task related to a `@frontity/types` refactor or something similar? I don't want these suggestions to disappear into the void. :eyes:

Reply:
No, not really, good suggestion. I will open one as soon as I review them in the https://github.com/frontity/frontity/pull/415 PR 🙂👍